src/compiler/scala/tools/nsc/backend/jvm/opt/ByteCodeRepository.scala

```diff
@@ -35,68 +35,33 @@ abstract class ByteCodeRepository {
    */
   val compilingClasses: concurrent.Map[InternalName, (ClassNode, String)] = recordPerRunCache(concurrent.TrieMap.empty)
 
+  private val maxCacheSize = 1500
+
   /**
-   * Cache for parsed ClassNodes.
-   * The `Long` field encodes the age of the node in the map, which allows removing old entries when
-   * the map grows too large (see limitCacheSize).
+   * Cache for parsed ClassNodes, an LRU or size maxCacheSize
    * For Java classes in mixed compilation, the map contains an error message: no ClassNode is
    * generated by the backend and also no classfile that could be parsed.
    */
-  val parsedClasses: concurrent.Map[InternalName, Either[ClassNotFound, (ClassNode, Long)]] = recordPerRunCache(concurrent.TrieMap.empty)
+  val parsedClasses: mutable.Map[InternalName, Either[ClassNotFound, ClassNode]] = recordPerRunCache(LruMap[InternalName, Either[ClassNotFound, ClassNode]](maxCacheSize, true))
 
   /**
    * Contains the internal names of all classes that are defined in Java source files of the current
    * compilation run (mixed compilation). Used for more detailed error reporting.
    */
   val javaDefinedClasses: mutable.Set[InternalName] = recordPerRunCache(mutable.Set.empty)
 
-  private val maxCacheSize = 1500
-  private val targetSize   = 500
 
   def initialize(): Unit = {
     javaDefinedClasses ++= frontendAccess.javaDefinedClasses
   }
 
-  private object lruCounter extends AtomicLong(0l) with collection.generic.Clearable {
-    def clear(): Unit = { this.set(0l) }
-  }
-  recordPerRunCache(lruCounter)
-
-  /**
-   * Prevent the code repository from growing too large. Profiling reveals that the average size
-   * of a ClassNode is about 30 kb. I observed having 17k+ classes in the cache, i.e., 500 mb.
```

Comment:
could you move this comment up to the definition of `private val maxCacheSize`?

Reply:
done in e28ad86